src/main/java/eu/interop/federationgateway/validator/DiagnosisKeyBatchValidator.java

```diff
@@ -15,24 +17,59 @@
   @Override
   public boolean isValid(EfgsProto.DiagnosisKeyBatch diagnosisKeyBatch, ConstraintValidatorContext context) {
 
+    /*
+     see
+       - https://github.com/google/exposure-notifications-server/blob/5a77e6982b16c7282c77dea39772491cc7b7dc8b/internal/publish/model/exposure_model.go#L338
+       - https://github.com/google/exposure-notifications-server/blob/5a77e6982b16c7282c77dea39772491cc7b7dc8b/tools/export-analyzer/main.go#L123
+     */
+
+    long minimumRollingStart = Instant
+      .now()
+      .truncatedTo(ChronoUnit.DAYS)
+      .minus(15, ChronoUnit.DAYS)
+      .getEpochSecond() / 600;
+
+    long maximumRollingStart = Instant
+      .now()
+      .getEpochSecond() / 600;
+    maximumRollingStart += 1;
+
     List<EfgsProto.DiagnosisKey> diagnosisKeys = diagnosisKeyBatch.getKeysList();
     for (EfgsProto.DiagnosisKey diagnosisKey : diagnosisKeys) {
-      if (diagnosisKey.getReportType() == EfgsProto.ReportType.UNKNOWN) {
-        log.error(VALIDATION_FAILED_MESSAGE + "Invalid report-type.");
-        return false;
-      } else if (diagnosisKey.getKeyData() == null || diagnosisKey.getKeyData().isEmpty()) {
-        log.error(VALIDATION_FAILED_MESSAGE + "The keydata is empty or null.");
-        return false;
-      } else if (diagnosisKey.getRollingPeriod() == 0) {
-        log.error(VALIDATION_FAILED_MESSAGE + "Invalid rolling period.");
-        return false;
-      } else if (diagnosisKey.getTransmissionRiskLevel() == 0 || diagnosisKey.getTransmissionRiskLevel() > 8) {
-        log.error(VALIDATION_FAILED_MESSAGE + "Invalid transmission risk level.");
-        return false;
-      }
+      if (diagnosisKey.getKeyData() == null || diagnosisKey.getKeyData().isEmpty()) {
+        return fail("The keydata is empty or null.", context);
+
+      } else if (diagnosisKey.getKeyData().size() != 16) {
+        return fail("The keydata is not 16 bytes.", context);
+
+      } else if (diagnosisKey.getRollingStartIntervalNumber() < minimumRollingStart
```

Comment:
in our experience w/ working app developers - this gets kind of sloppy.

we implement partial success in the case of keys that are too old combined w/ some valid keys.

I understand the EFGS might not be able to do that w/ signed batches.

Reply:
@SchulzeS and I already thought about removing this validation. So @SchulzeS, do we wanna remove it finally ?

Reply:
If a country has keys which are too old it will get the information that the batch was rejected because of that.
Then the country just has to remove the old keys from batch and sign and send again.

In my opinion we should not remove this validation because otherwise it would be possible to (accidently) upload keys with bad timestamps